version: "3.9"
services:
    db:
        image: postgres
        environment:
            - POSTGRES_USER=${POSTGRES_USER}
            - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
            - POSTGRES_DB=shopify-challenge-db
            - PGADMIN_LISTEN_PORT=5432
        user: root
        volumes:
            - postgres:/bar/lib/postgressql/data
        ports:
            - "5432:5432"
        healthcheck:
            test: ["CMD-SHELL", "pg_isready -U postgres"]
            interval: 5s
            timeout: 5s
            retries: 5

    web:
        build: .
        ports:
            - "8080:8080"
        depends_on:
            db:
                condition: service_healthy

    pgadmin:
        container_name: pgadmin4
        image: dpage/pgadmin4
        environment:
            PGADMIN_DEFAULT_EMAIL: admin@example.com
            PGADMIN_DEFAULT_PASSWORD: admin
            PGADMIN_CONFIG_SERVER_MODE: "False"
            PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: "False"
        volumes:
            - ./pgadmin/data:/root/.pgadmin
            - ./docker_pgadmin_servers.json:/pgadmin4/servers.json
        ports:
            - "80:80"
        entrypoint:
            - "/bin/sh"
            - "-c"
            - "/bin/echo 'db:5432:*:${POSTGRES_USER}:${POSTGRES_PASSWORD}' > /tmp/pgpassfile && chmod 600 /tmp/pgpassfile && /entrypoint.sh"
        depends_on: 
            - db
        restart: unless-stopped
        profiles: ["dev"]

volumes:
    postgres:
        name: postgres-data